writer.go

```diff
@@ -704,8 +712,29 @@ func (w *writer) write(conn *Conn, batch []Message, resch [](chan<- error)) (ret
 
 	t0 := time.Now()
 	conn.SetWriteDeadline(time.Now().Add(w.writeTimeout))
-	if _, err = conn.WriteCompressedMessages(w.codec, batch...); err != nil {
-		w.stats.errors.observe(1)
+	attempts := 0
+	for w.retries > attempts {
+		if _, err = conn.WriteCompressedMessages(w.codec, batch...); err != nil {
+			w.stats.errors.observe(1)
+			switch err {
+			case DuplicateSequenceNumber:
+				// we've moved beyond the batch but havn't noticed it. Just return sucess.
+				// See https://github.com/apache/kafka/blob/trunk/clients/src/main/java/org/apache/kafka/clients/producer/internals/Sender.java#L617
+				break
+			case TopicAuthorizationFailed, ClusterAuthorizationFailed, GroupAuthorizationFailed:
+				err = fmt.Errorf("auth error writing messages to %s (partition %d): %s", w.topic, w.partition, err)
+				break
+
+			default:
+				attempts++
+				err = fmt.Errorf("error writing messages to %s (partition %d): %s", w.topic, w.partition, err)
+				backoff(attempts, 5*time.Millisecond, 100*time.Millisecond)
```

Comment:
They have constants `backoffDelayMin` and `backoffDelayMax`

Reply:
Should they be configurable?

Reply:
Would we change them? I kinda doubt we'd ever need different values.

Reply:
Yeah, maybe we should add some consts for this. 
But I don't want to delay an entire second, I want a bit faster retries

Reply:
yeah, there is a TODO to make they configurable as well